src/Medidata.ZipkinTracer.Core/SpanTracer.cs

```diff
@@ -81,11 +81,11 @@ public virtual Span SendClientSpan(string requestName, string traceId, string pa
             return newSpan;
         }
 
-        public virtual void ReceiveClientSpan(Span span, string clientServiceName)
+        public virtual void ReceiveClientSpan(Span span)
         {
             var annotation = new Annotation()
             {
-                Host = zipkinEndpoint.GetEndpoint(clientServiceName),
+                Host = span.Annotations[0].Host,
```

Comment:
@lschreck-mdsol Thanks for reviewing the implementation. ReceiveClientSpan (end client trace) method is expecting a span object which was created prior to this call (on start of the client trace). As from what we understand, It expects Annotations list populated with start trace annotation. 

In the event 
if null span --> it is already handled on the caller of this method, please check ZipkinClient.cs, 
if null Annotations, empty Annotation collection cases --> Assumed that span data comes from start of client trace and expected to have Annotations and at least have one element. In case if it is invalid data, it will not submit the log. It would raise exception and will be caught in ZipkinClient.cs

Thank you.